projects/core/components/data-list/data-list.component.ts

```diff
@@ -44,6 +42,15 @@ export function tuiInjectDataListSize(): TuiSizeL | TuiSizeS {
     providers: [tuiAsDataListAccessor(TuiDataListComponent)],
     host: {
         role: 'listbox',
+        '[attr.data-size]': 'size',
+        '(focusin)': 'onFocusIn($event.relatedTarget, $event.currentTarget)',
+        '(mousedown.prevent)': 'noop()',
+        '(wheel.silent.passive)': 'handleFocusLossIfNecessary()',
+        '(mouseleave)': 'handleFocusLossIfNecessary($event.target)',
+        '(keydown.tab)': 'handleFocusLossIfNecessary($event.target)',
+        '(keydown.shift.tab)': 'handleFocusLossIfNecessary($event.target)',
```

Comment:
According to the previous version, `$event.target` is required only for `mouseleave`

Reply:
fixed